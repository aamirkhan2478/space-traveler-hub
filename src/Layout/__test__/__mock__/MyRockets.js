import React from "react";
import { ListGroup } from "react-bootstrap";
import { Provider } from "react-redux";
import store from "../../../redux/configureStore";

const MyRockets = () => {
  const rockets = [
    {
      rocket_name: "Falcon 9",
      reserved: true,
    },
  ];
  return (
    <Provider store={store}>
      <h3>My Rockets</h3>
      <ListGroup>
        {rockets
          .slice(1)
          .map(
            (rocket) =>
              rocket.reserved && (
                <ListGroup.Item>{rocket.rocket_name}</ListGroup.Item>
              )
          )}
      </ListGroup>
    </Provider>
  );
};

export default MyRockets;
