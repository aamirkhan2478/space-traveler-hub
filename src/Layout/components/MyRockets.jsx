import React from 'react';
import { ListGroup } from 'react-bootstrap';
import { useSelector } from 'react-redux';

const MyRockets = () => {
  const { rockets } = useSelector((state) => state.rocket);
  return (
    <>
      <h3>My Rockets</h3>
      <ListGroup>
        {rockets.map(
          (rocket) => rocket.reserved && (
          <ListGroup.Item>{rocket.rocket_name}</ListGroup.Item>
          ),
        )}
      </ListGroup>
    </>
  );
};

export default MyRockets;
