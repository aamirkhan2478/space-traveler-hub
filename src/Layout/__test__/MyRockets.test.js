import renderer from 'react-test-renderer';
import MyRockets from './__mock__/MyRockets';

it('renders rockets correctly', () => {
  const tree = renderer.create(<MyRockets />).toJSON();
  expect(tree).toMatchSnapshot();
});
