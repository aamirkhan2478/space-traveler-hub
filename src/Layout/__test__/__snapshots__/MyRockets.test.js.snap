// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders rockets correctly 1`] = `
Array [
  <h3>
    My Rockets
  </h3>,
  <div
    className="list-group"
    onKeyDown={[Function]}
  />,
]
`;
